src/modules/github.ts

```diff
@@ -0,0 +1,281 @@
+
+  const params = new URLSearchParams({
+    client_id: process.env.CLIENT_ID,
+    redirect_uri: chrome.identity.getRedirectURL(),
+    scope: "repo user:email",
+  });
+  const options = {
+    url: `https://github.com/login/oauth/authorize?${params}`,
+    interactive: true,
+  };
+  const responseUrl = await chrome.identity.launchWebAuthFlow(options);
+  const code = new URL(responseUrl).searchParams.get("code");
+  if (!code) {
+    await errorReport({
+      message: `code not found in response: ${responseUrl}`,
+      context: "github.requestIdentity",
+    });
+  }
+
+  return code;
+};
+
+/**
+ * Exchange the code for an access token
+ * @param {string} code
+ * @returns {Promise<string>} access token
+ */
+const exchangeCodeForToken = async (code: string): Promise<string> => {
+  await debugReport({
+    message: "Exchange the code for an access token",
+    context: "github.exchangeCodeForToken",
+  });
+
+  if (!process.env.CLIENT_ID) {
+    await errorReport({
+      message: "CLIENT_ID not set",
+      context: "github.exchangeCodeForToken",
+    });
+  }
+
+  if (!process.env.CLIENT_SECRET) {
+    await errorReport({
+      message: "CLIENT_SECRET not set",
+      context: "github.exchangeCodeForToken",
+    });
+  }
+
+  const response = await fetch("https://github.com/login/oauth/access_token", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+    },
+    body: JSON.stringify({
+      client_id: process.env.CLIENT_ID,
+      client_secret: process.env.CLIENT_SECRET,
+      code,
+    }),
+  });
+
+  if (!response.ok) {
+    await errorReport({
+      message: `Unexpected status code: ${response.status}`,
+      context: "github.exchangeCodeForToken",
+    });
+  }
+
+  const payload = await response.json();
+  return payload.access_token as string;
+};
+
+/**
+ * Create a new GitHub repository if it doesn't exist
+ * @param {Octokit} octokit
+ */
+const createRepository = async (octokit: Octokit): Promise<void> => {
+  await debugReport({
+    message: "Create a new GitHub repository if it doesn't exist",
+    context: "github.createRepository",
+  });
+
+  const {
+    data: { login: username },
+  } = await octokit.rest.users.getAuthenticated();
+
+  const { data } = await octokit.rest.repos.listForUser({ username });
+  let repository = data.find((repo) => repo.name === "LeetCode");
+
+  if (!repository) {
+    await octokit.rest.repos.createUsingTemplate({
+      template_owner: "SyncLeet",
+      template_repo: "template",
+      name: "LeetCode",
+      description: "Sync: LeetCode -> GitHub",
+      private: false,
+    });
+    while (!repository) {
+      await new Promise((resolve) => setTimeout(resolve, 2500));
+      const { data } = await octokit.rest.repos.listForUser({ username });
+      repository = data.find((repo) => repo.name === "LeetCode");
+    }
+    await notifyReport({
+      message: "Created a new GitHub repository for LeetCode!",
```

Comment:
I didn't receive any notification for new repo when testing, did this happen to you?